src/libraries/System.Security.Cryptography.Cose/tests/CoseHeaderMapTests.cs

```diff
@@ -0,0 +1,248 @@
+
+            var writer = new CborWriter();
+            int currentHeader = KnownHeaders.Alg;
+            foreach ((CoseHeaderLabel Label, ReadOnlyMemory<byte> EncodedValue) in map)
+            {
+                if (currentHeader == KnownHeaders.Crit) // TODO remove when Crit gets supported
+                {
+                    currentHeader++;
+                }
+
+                Assert.Equal(new CoseHeaderLabel(currentHeader), Label);
+                ReadOnlyMemory<byte> ExpectedValue = currentHeader switch
+                {
+                    KnownHeaders.Alg => EncodeInt32((int)ECDsaAlgorithm.ES256, writer),
+                    KnownHeaders.Crit => GetDummyCritHeaderValue(),
+                    KnownHeaders.ContentType => EncodeString(ContentTypeDummyValue, writer),
+                    KnownHeaders.Kid => EncodeBytes(s_SampleContent, writer),
+                    KnownHeaders.IV or KnownHeaders.PartialIV => EncodeBytes(ReadOnlySpan<byte>.Empty, writer),
+                    _ => throw new InvalidOperationException()
+                };
+                Assert.True(ExpectedValue.Span.SequenceEqual(EncodedValue.Span));
+                currentHeader++;
+            }
+            Assert.Equal(KnownHeaders.PartialIV + 1, currentHeader);
+
+            static ReadOnlyMemory<byte> EncodeInt32(int value, CborWriter writer)
+            {
+                writer.WriteInt32(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeString(string value, CborWriter writer)
+            {
+                writer.WriteTextString(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeBytes(ReadOnlySpan<byte> value, CborWriter writer)
+            {
+                writer.WriteByteString(value);
+                return EncodeAndReset(writer);
+            }
+
+            static ReadOnlyMemory<byte> EncodeAndReset(CborWriter writer)
+            {
+                ReadOnlyMemory<byte> encodedValue = writer.Encode();
+                writer.Reset();
+                return encodedValue;
+            }
+        }
+
+        [Fact]
+        public void DecodedProtectedMapShouldBeReadOnly()
+        {
+            byte[] encodedMessage = CoseSign1Message.Sign(s_SampleContent, ECDsaKeys[ECDsaAlgorithm.ES256], HashAlgorithmName.SHA256);
+            CoseSign1Message message = CoseMessage.DecodeSign1(encodedMessage);
+            Assert.True(message.ProtectedHeader.IsReadOnly);
+        }
+
+        [Fact]
+        public void GetValueFromReadOnlyProtectedMap()
+        {
+            byte[] encodedMessage = CoseSign1Message.Sign(s_SampleContent, ECDsaKeys[ECDsaAlgorithm.ES256], HashAlgorithmName.SHA256);
+            CoseSign1Message message = CoseMessage.DecodeSign1(encodedMessage);
+            Assert.True(message.ProtectedHeader.IsReadOnly);
+
+            int expectedAlgorithm = (int)ECDsaAlgorithm.ES256;
```

Comment:
Again, by reusing a const from the source this test is virtually guaranteed to pass.  Instead, the test should assert that it's -7, since that checks that our implementation matches the specification (from a black-box perspective)

Reply:
(Since the name of this test is that it's proving a read works on an IsReadOnly map, pretend I left this comment in a place where we're checking that the algorithm ID we wrote down was correct)